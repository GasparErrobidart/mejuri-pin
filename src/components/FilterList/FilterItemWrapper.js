import styled from 'styled-components'

const FilterItemWrapper = styled.div`
  display: flex;
  padding-left: 30px;
  &:hover,&:active{
    border-color: ${({theme})=> theme.colors.middleGray};
  }
  cursor:pointer;

  & *{
    transition: all 200ms ease-in;
    color : ${({checked,theme}) => checked ? theme.colors.darkGray : theme.colors.darkMiddleGray};
  }

  &:hover *{
    color : ${({checked,theme}) => theme.colors.darkGray };
  }

  & div:first-child{
    overflow: hidden;
    flex: 0 0 ${({checked}) => checked ? "25px" : 0};
    height: 15px;
  }
`

export default FilterItemWrapper
