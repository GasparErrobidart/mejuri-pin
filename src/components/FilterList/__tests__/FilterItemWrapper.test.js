import '@testing-library/jest-dom'
import 'jest-styled-components'

import React                 from 'react'
import {ThemeProvider}       from 'styled-components'
import { cleanup }           from '@testing-library/react'
import renderer              from 'react-test-renderer'

import theme                 from '../../GlobalStyle/themes/mejuri-pin'

import FilterItemWrapper   from '../FilterItemWrapper'


afterEach(cleanup)

test('Renders "FilterItemWrapper" component', () => {
  const testText        = "FilterItemWrapper Test Text"
  const component       = renderer.create(
    <ThemeProvider theme={theme}>
      <FilterItemWrapper>{testText}</FilterItemWrapper>
    </ThemeProvider>);
  const tree            = component.toJSON()
  expect(tree).toMatchSnapshot();
});
