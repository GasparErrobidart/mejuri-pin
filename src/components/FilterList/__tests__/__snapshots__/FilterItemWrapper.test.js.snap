// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders "FilterItemWrapper" component 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-left: 30px;
  cursor: pointer;
}

.c0:hover,
.c0:active {
  border-color: #CCCCCC;
}

.c0 * {
  -webkit-transition: all 200ms ease-in;
  transition: all 200ms ease-in;
  color: #888888;
}

.c0:hover * {
  color: #303030;
}

.c0 div:first-child {
  overflow: hidden;
  -webkit-flex: 0 0 0;
  -ms-flex: 0 0 0;
  flex: 0 0 0;
  height: 15px;
}

<div
  className="c0"
>
  FilterItemWrapper Test Text
</div>
`;
